pkg/kubelet/dockertools/manager.go

```diff
@@ -1362,6 +1363,15 @@ func (dm *DockerManager) killContainer(containerID kubecontainer.ContainerID, co
 	return err
 }
 
+func (dm *DockerManager) generateContainerEvent(containerID kubecontainer.ContainerID, podName, reason, message string) {
+	ref, ok := dm.containerRefManager.GetRef(containerID)
+	if !ok {
+		glog.Warningf("No ref for pod '%q'", podName)
+		return
+	}
+	dm.recorder.Event(ref, api.EventTypeNormal, reason, message)
```

Comment:
Good catch. Changed to `generateFailedContainerEvent()` to make it re-usable for other scenarios.